community/detectors/roxy_wi_cve_2022_31137/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202231137/Cve202231137Detector.java

```diff
@@ -0,0 +1,208 @@
+
+  private final PayloadGenerator payloadGenerator;
+
+  @Inject
+  Cve202231137Detector(
+      @UtcClock Clock utcClock, HttpClient httpClient, PayloadGenerator payloadGenerator) {
+
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient);
+    this.payloadGenerator = checkNotNull(payloadGenerator);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("Cve202231137Detector starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isRoxyWiWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isRoxyWiWebService(NetworkService networkService) {
+    // note that this fingerprint phase is only for vulnerable versions,
+    // because newer Roxy-Wi versions have different endpoints for login
+    String targetUrl =
+        NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + "app/login.py";
+    try {
+      Optional<String> response =
+          httpClient.send(get(targetUrl).withEmptyHeaders().build(), networkService).bodyString();
+      return response.isPresent() && response.get().contains("<title>Login page - Roxy-WI</title>");
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Request to target %s failed", networkService);
+    }
+    return false;
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    return (payloadGenerator.isCallbackServerEnabled() && isVulnerableWithCallback(networkService))
+        || isVulnerableWithoutCallback(networkService);
+  }
+
+  private boolean isVulnerableWithoutCallback(NetworkService networkService) {
+    PayloadGeneratorConfig config =
+        PayloadGeneratorConfig.newBuilder()
+            .setVulnerabilityType(PayloadGeneratorConfig.VulnerabilityType.REFLECTIVE_RCE)
+            .setInterpretationEnvironment(
+                PayloadGeneratorConfig.InterpretationEnvironment.LINUX_SHELL)
+            .setExecutionEnvironment(
+                PayloadGeneratorConfig.ExecutionEnvironment.EXEC_INTERPRETATION_ENVIRONMENT)
+            .build();
+
+    Payload payload = payloadGenerator.generate(config);
+    String cmd = payload.getPayload();
+
+    HttpResponse response =
+        sendRequest(
+            networkService,
+            String.format(HTTP_PARAMETERS, URLEncoder.encode(cmd, StandardCharsets.UTF_8)));
+    if (response != null) {
+      if (response.bodyString().isEmpty()) {
+        return false;
+      }
+      return payload.checkIfExecuted(response.bodyString().get());
+    } else return false;

```

Comment:
These checks can be simplified as follows:

```suggestion
    if (response == null || response.bodyString().isEmpty()) {
      return false;
    }
  
    return payload.checkIfExecuted(response.bodyString().get());
```